lookup_host result can't be send across threads safely

Hey,
I'm not 100% sure why this fails, but when trying to do the following I receive the error:
Error:
```
error[E0277]: `dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>` cannot be sent between threads safely
  --> examples\lookup-runtime.rs:13:13
   |
13 |     runtime.spawn(real_main());
   |             ^^^^^ `dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>` cannot be sent between threads safely
   |
   = help: the trait `Send` is not implemented for `dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>`
   = note: required because of the requirements on the impl of `Send` for `Unique<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>`
   = note: required because it appears within the type `Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>`
   = note: required because it appears within the type `Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>`
   = note: required because it appears within the type `tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>`
   = note: required because it appears within the type `(tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>, tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>)`
   = note: required because it appears within the type `for<'r> {ResumeTy, &'r &StubResolver, domain::base::Dname<Vec<u8>>, (tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>, tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>), [closure@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0}::{closure#0}], tokio::future::poll_fn::PollFn<[closure@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0}::{closure#0}]>, ()}`
   = note: required because it appears within the type `[static generator@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0} for<'r> {ResumeTy, &'r &StubResolver, domain::base::Dname<Vec<u8>>, (tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>, tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>), [closure@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0}::{closure#0}], tokio::future::poll_fn::PollFn<[closure@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0}::{closure#0}]>, ()}]`
   = note: required because it appears within the type `from_generator::GenFuture<[static generator@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0} for<'r> {ResumeTy, &'r &StubResolver, domain::base::Dname<Vec<u8>>, (tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>, tokio::future::maybe_done::MaybeDone<Pin<Box<dyn Future<Output = std::result::Result<domain::resolv::stub::Answer, std::io::Error>>>>>), [closure@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0}::{closure#0}], tokio::future::poll_fn::PollFn<[closure@domain::resolv::lookup::lookup_host<&StubResolver, domain::base::Dname<Vec<u8>>>::{closure#0}::{closure#0}]>, ()}]>`
   = note: required because it appears within the type `impl Future`
   = note: required because it appears within the type `impl Future`
   = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2> {ResumeTy, &'r StubResolver, domain::base::Dname<Vec<u8>>, &'s &'t0 StubResolver, impl Future, ()}`
   = note: required because it appears within the type `[static generator@StubResolver::lookup_host<domain::base::Dname<Vec<u8>>>::{closure#0} for<'r, 's, 't0, 't1, 't2> {ResumeTy, &'r StubResolver, domain::base::Dname<Vec<u8>>, &'s &'t0 StubResolver, impl Future, ()}]`
   = note: required because it appears within the type `from_generator::GenFuture<[static generator@StubResolver::lookup_host<domain::base::Dname<Vec<u8>>>::{closure#0} for<'r, 's, 't0, 't1, 't2> {ResumeTy, &'r StubResolver, domain::base::Dname<Vec<u8>>, &'s &'t0 StubResolver, impl Future, ()}]>`
   = note: required because it appears within the type `impl Future`
   = note: required because it appears within the type `impl Future`
   = note: required because it appears within the type `for<'r, 's> {ResumeTy, StubResolver, domain::base::Dname<Vec<u8>>, impl Future, ()}`
   = note: required because it appears within the type `[static generator@examples\lookup-runtime.rs:16:22: 21:2 for<'r, 's> {ResumeTy, StubResolver, domain::base::Dname<Vec<u8>>, impl Future, ()}]`
   = note: required because it appears within the type `from_generator::GenFuture<[static generator@examples\lookup-runtime.rs:16:22: 21:2 for<'r, 's> {ResumeTy, StubResolver, domain::base::Dname<Vec<u8>>, impl Future, ()}]>`
   = note: required because it appears within the type `impl Future`
   = note: required because it appears within the type `impl Future`
```

Example:
```
use domain::base::name::UncertainDname;
use domain::resolv::StubResolver;
use domain::base::Dname;
use std::str::FromStr;

fn main() {
    let runtime = tokio::runtime::Builder::new_current_thread()
        .enable_all()
        .build().expect("failed to create runtime");

    runtime.spawn(real_main());
}

async fn real_main() {
    let resolver = StubResolver::new();
    let address = Dname::<Vec<u8>>::from_str("example.com").expect("failed to parse test address");
    resolver.lookup_host(address).await;
}
```

I'm not sure if I'm overlooking anything.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "domain"
-version = "0.6.2-dev"
+version = "0.7.0-dev"
 edition = "2018"
 authors = ["Martin Hoffmann <martin@nlnetlabs.nl>"]
 description = "A DNS library for Rust."

--- a/src/resolv/resolver.rs
+++ b/src/resolv/resolver.rs
@@ -25,7 +25,7 @@ pub trait Resolver {
     type Answer: AsRef<Message<Self::Octets>>;
 
     /// The future resolving into an answer.
-    type Query: Future<Output = Result<Self::Answer, io::Error>>;
+    type Query: Future<Output = Result<Self::Answer, io::Error>> + Send;
 
     /// Returns a future answering a question.
     ///

--- a/src/resolv/stub/mod.rs
+++ b/src/resolv/stub/mod.rs
@@ -204,7 +204,7 @@ impl<'a> Resolver for &'a StubResolver {
     type Octets = Bytes;
     type Answer = Answer;
     type Query =
-        Pin<Box<dyn Future<Output = Result<Answer, io::Error>> + 'a>>;
+        Pin<Box<dyn Future<Output = Result<Answer, io::Error>> + Send + 'a>>;
 
     fn query<N, Q>(&self, question: Q) -> Self::Query
     where

